diff --git a/src/alejandra_cli/src/cli.rs b/src/alejandra_cli/src/cli.rs
index bab102c..b90bf1d 100644
--- a/src/alejandra_cli/src/cli.rs
+++ b/src/alejandra_cli/src/cli.rs
@@ -7,7 +7,6 @@ use futures::future::RemoteHandle;
 use futures::stream::FuturesUnordered;
 use futures::task::SpawnExt;
 
-use crate::ads::random_ad;
 use crate::verbosity::Verbosity;
 
 /// The Uncompromising Nix Code Formatter.
@@ -203,11 +202,6 @@ pub fn main() -> std::io::Result<()> {
                     (true, false) => "requires formatting",
                 }
             );
-
-            if in_place {
-                eprintln!();
-                eprint!("{}", random_ad());
-            }
         }
 
         std::process::exit(if in_place { 0 } else { 2 });
@@ -218,8 +212,6 @@ pub fn main() -> std::io::Result<()> {
         eprintln!(
             "Congratulations! Your code complies with the Alejandra style."
         );
-        eprintln!();
-        eprint!("{}", random_ad());
     }
 
     std::process::exit(0);
diff --git a/src/alejandra_cli/src/lib.rs b/src/alejandra_cli/src/lib.rs
index fd49ce8..258f656 100644
--- a/src/alejandra_cli/src/lib.rs
+++ b/src/alejandra_cli/src/lib.rs
@@ -1,4 +1,3 @@
-mod ads;
 pub mod cli;
 mod find;
 mod verbosity;
